<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>640</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="GLWidget" name="glWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>580</width>
      <height>580</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="openButton">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>10</y>
      <width>190</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="locale">
     <locale language="English" country="UnitedStates"/>
    </property>
    <property name="text">
     <string> Open File</string>
    </property>
    <property name="icon">
     <iconset theme="QIcon::ThemeIcon::DocumentOpen">
      <normaloff>.</normaloff>.</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>12</width>
      <height>12</height>
     </size>
    </property>
    <property name="shortcut">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="paintButton">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>600</y>
      <width>190</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>12</pointsize>
      <bold>false</bold>
      <kerning>true</kerning>
     </font>
    </property>
    <property name="locale">
     <locale language="English" country="UnitedStates"/>
    </property>
    <property name="text">
     <string> Paint</string>
    </property>
    <property name="icon">
     <iconset theme="QIcon::ThemeIcon::MailMessageNew">
      <normaloff>.</normaloff>.</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>12</width>
      <height>12</height>
     </size>
    </property>
    <property name="shortcut">
     <string/>
    </property>
   </widget>
   <widget class="QToolBox" name="glWidgetMenu">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>110</y>
      <width>190</width>
      <height>451</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="acceptDrops">
     <bool>false</bool>
    </property>
    <property name="toolTipDuration">
     <number>-1</number>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <property name="tabSpacing">
     <number>4</number>
    </property>
    <widget class="QWidget" name="Transform">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>190</width>
       <height>395</height>
      </rect>
     </property>
     <attribute name="icon">
      <iconset theme="QIcon::ThemeIcon::MailAttachment">
       <normaloff>.</normaloff>.</iconset>
     </attribute>
     <attribute name="label">
      <string>Transform</string>
     </attribute>
     <widget class="QPushButton" name="rotateButtonX">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>250</y>
        <width>50</width>
        <height>30</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="locale">
       <locale language="English" country="UnitedStates"/>
      </property>
      <property name="text">
       <string>▲ X</string>
      </property>
      <property name="iconSize">
       <size>
        <width>12</width>
        <height>12</height>
       </size>
      </property>
      <property name="shortcut">
       <string/>
      </property>
     </widget>
     <widget class="QPushButton" name="leftButton">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>40</y>
        <width>40</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>◄</string>
      </property>
     </widget>
     <widget class="QPushButton" name="upButton">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>40</y>
        <width>40</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>▲</string>
      </property>
     </widget>
     <widget class="QPushButton" name="rightButton">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>40</y>
        <width>40</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>►</string>
      </property>
     </widget>
     <widget class="QPushButton" name="downButton">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>40</y>
        <width>40</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>▼</string>
      </property>
     </widget>
     <widget class="QPushButton" name="closerButton">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>80</y>
        <width>40</width>
        <height>30</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>15</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>+</string>
      </property>
     </widget>
     <widget class="QPushButton" name="furtherButton">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>80</y>
        <width>40</width>
        <height>30</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>15</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>-</string>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>0</y>
        <width>170</width>
        <height>30</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>12</pointsize>
        <bold>true</bold>
        <underline>true</underline>
       </font>
      </property>
      <property name="text">
       <string>_________SHIFT_________</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>170</width>
        <height>30</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>12</pointsize>
        <bold>true</bold>
        <underline>true</underline>
       </font>
      </property>
      <property name="text">
       <string>________SCALE________</string>
      </property>
     </widget>
     <widget class="QPushButton" name="increaseButton">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>160</y>
        <width>90</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="icon">
       <iconset theme="QIcon::ThemeIcon::ZoomIn">
        <normaloff>.</normaloff>.</iconset>
      </property>
     </widget>
     <widget class="QPushButton" name="decreaseButton">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>160</y>
        <width>90</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="icon">
       <iconset theme="QIcon::ThemeIcon::ZoomOut">
        <normaloff>.</normaloff>.</iconset>
      </property>
     </widget>
     <widget class="QPushButton" name="rotateButtonZ">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>250</y>
        <width>50</width>
        <height>30</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="locale">
       <locale language="English" country="UnitedStates"/>
      </property>
      <property name="text">
       <string> Z</string>
      </property>
      <property name="icon">
       <iconset theme="QIcon::ThemeIcon::ViewRefresh">
        <normaloff>.</normaloff>.</iconset>
      </property>
      <property name="iconSize">
       <size>
        <width>12</width>
        <height>12</height>
       </size>
      </property>
      <property name="shortcut">
       <string/>
      </property>
     </widget>
     <widget class="QPushButton" name="rotateButtonY">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>250</y>
        <width>50</width>
        <height>30</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="locale">
       <locale language="English" country="UnitedStates"/>
      </property>
      <property name="text">
       <string>◄ Y</string>
      </property>
      <property name="iconSize">
       <size>
        <width>12</width>
        <height>12</height>
       </size>
      </property>
      <property name="shortcut">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>210</y>
        <width>170</width>
        <height>30</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>12</pointsize>
        <bold>true</bold>
        <underline>true</underline>
       </font>
      </property>
      <property name="text">
       <string>________ROTATE________</string>
      </property>
     </widget>
     <widget class="QPushButton" name="rotateButtonX_2">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>290</y>
        <width>50</width>
        <height>30</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="locale">
       <locale language="English" country="UnitedStates"/>
      </property>
      <property name="text">
       <string>▼ X</string>
      </property>
      <property name="iconSize">
       <size>
        <width>12</width>
        <height>12</height>
       </size>
      </property>
      <property name="shortcut">
       <string/>
      </property>
     </widget>
     <widget class="QPushButton" name="rotateButtonY_2">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>290</y>
        <width>50</width>
        <height>30</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="locale">
       <locale language="English" country="UnitedStates"/>
      </property>
      <property name="text">
       <string>► Y</string>
      </property>
      <property name="iconSize">
       <size>
        <width>12</width>
        <height>12</height>
       </size>
      </property>
      <property name="shortcut">
       <string/>
      </property>
     </widget>
     <widget class="QPushButton" name="rotateButtonZ_2">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>290</y>
        <width>50</width>
        <height>30</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="locale">
       <locale language="English" country="UnitedStates"/>
      </property>
      <property name="text">
       <string> Z</string>
      </property>
      <property name="icon">
       <iconset theme="QIcon::ThemeIcon::SystemReboot">
        <normaloff>.</normaloff>.</iconset>
      </property>
      <property name="iconSize">
       <size>
        <width>12</width>
        <height>12</height>
       </size>
      </property>
      <property name="shortcut">
       <string/>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="Properties">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>190</width>
       <height>395</height>
      </rect>
     </property>
     <property name="contextMenuPolicy">
      <enum>Qt::NoContextMenu</enum>
     </property>
     <attribute name="icon">
      <iconset theme="QIcon::ThemeIcon::DocumentProperties">
       <normaloff>.</normaloff>.</iconset>
     </attribute>
     <attribute name="label">
      <string>Properties</string>
     </attribute>
     <widget class="QPushButton" name="applyButton">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>350</y>
        <width>190</width>
        <height>30</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="locale">
       <locale language="English" country="UnitedStates"/>
      </property>
      <property name="text">
       <string> Apply</string>
      </property>
      <property name="icon">
       <iconset theme="appointment-new">
        <normaloff>.</normaloff>.</iconset>
      </property>
      <property name="iconSize">
       <size>
        <width>12</width>
        <height>12</height>
       </size>
      </property>
      <property name="shortcut">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="backgroundLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>270</y>
        <width>170</width>
        <height>30</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>12</pointsize>
        <underline>true</underline>
       </font>
      </property>
      <property name="text">
       <string>BACKGROUND</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lineLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>170</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>14</pointsize>
        <underline>true</underline>
       </font>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="text">
       <string>LINE</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="pointLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>160</y>
        <width>170</width>
        <height>30</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>16</pointsize>
        <underline>true</underline>
       </font>
      </property>
      <property name="text">
       <string>POINT</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="lineLabel_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>100</y>
        <width>50</width>
        <height>20</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Color</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QComboBox" name="comboBoxLineColor">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>100</y>
        <width>110</width>
        <height>22</height>
       </rect>
      </property>
      <property name="editable">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QLabel" name="lineLabel_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>130</y>
        <width>51</width>
        <height>31</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="text">
       <string>Width</string>
      </property>
     </widget>
     <widget class="QSlider" name="sliderLineWidth">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>130</y>
        <width>81</width>
        <height>31</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QComboBox" name="comboBoxPointColor">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>220</y>
        <width>110</width>
        <height>22</height>
       </rect>
      </property>
      <property name="editable">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QLabel" name="pointLabel_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>240</y>
        <width>50</width>
        <height>20</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="text">
       <string>Size</string>
      </property>
     </widget>
     <widget class="QLabel" name="pointLabel_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>220</y>
        <width>50</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Color</string>
      </property>
     </widget>
     <widget class="QSlider" name="sliderPointSize">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>240</y>
        <width>70</width>
        <height>31</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="pointLabel_1">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>190</y>
        <width>50</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Type</string>
      </property>
     </widget>
     <widget class="QComboBox" name="comboBoxPointType">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>190</y>
        <width>110</width>
        <height>22</height>
       </rect>
      </property>
      <property name="editable">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>300</y>
        <width>40</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Color</string>
      </property>
     </widget>
     <widget class="QComboBox" name="comboBoxBackColor">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>300</y>
        <width>110</width>
        <height>22</height>
       </rect>
      </property>
      <property name="editable">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEditLineWidth">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>130</y>
        <width>30</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEditPointSize">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>250</y>
        <width>30</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>0</y>
        <width>171</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Arial</family>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="title">
       <string>PROJECTION TYPE</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <widget class="QRadioButton" name="radioTypeCentral">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>20</y>
         <width>81</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Central</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioTypeParallel">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>20</y>
         <width>91</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Parallel</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>171</width>
        <height>21</height>
       </rect>
      </property>
      <property name="title">
       <string/>
      </property>
      <widget class="QRadioButton" name="radioLineSolid">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>71</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Solid</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioLineDashed">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>0</y>
         <width>91</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Dashed</string>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="QLabel" name="pathFile">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>10</y>
      <width>380</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>16</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>File Name</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelVertex">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>35</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>16</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>ver</string>
    </property>
   </widget>
   <widget class="QLabel" name="labelPolygon">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>10</y>
      <width>35</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>16</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>pol</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="countVertex">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>10</y>
      <width>61</width>
      <height>30</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="countPolygon">
    <property name="geometry">
     <rect>
      <x>139</x>
      <y>10</y>
      <width>61</width>
      <height>30</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="projectionLabel">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>60</y>
      <width>170</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>12</pointsize>
      <underline>true</underline>
     </font>
    </property>
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;PROJECTION TYPE&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>glwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
